packages/gatsby-theme-newrelic/plugins/gatsby-source-swiftype/gatsby-node.js

```diff
@@ -0,0 +1,93 @@
+const fs = require('fs');
+const createRelatedResourceNode = require('./src/createRelatedResourceNode');
+const getRelatedResources = require('./src/getRelatedResources');
+
+const writeableData = {};
+
+exports.onPreBootstrap = (_, pluginOptions) => {
+  const { file } = pluginOptions;
+
+  if (!fs.existsSync(file)) {
+    fs.writeFileSync(file, '{}');
+  }
+};
+
+exports.onCreateNode = async (
+  { actions, node, getNodesByType, createNodeId, createContentDigest },
+  pluginOptions
+) => {
+  const { createNode, createParentChildLink } = actions;
+  const { filterNode = () => false, getPath } = pluginOptions;
```

Comment:
what does this `filterNode` function do? how does this inline function work

Reply:
It looks like this is being used as a default value in case `filterNode` isn't specified. If we use this plugin without `filterNode`, it will just be a function that always returns `false`.

Reply:
The reason that `filterNode` returns `false` rather than `true` is that we want to avoid executing searches on Swiftype for pages that aren't explicitly set as needing related resources. This helps us avoid the cost of executing a query against Swiftype for pages that we don't use this for.